demo/src/bootstrap/app.js

```diff
@@ -89,6 +90,49 @@ export default class App extends React.Component {
   updateConfig = config => {
     this.setState(config);
   };
+
+  handleFile = element => {
+    // https://www.html5rocks.com/en/tutorials/file/dndfiles/
+    const files = element.files;
+    if (!files.length) {
+      alert("Please select a file!");
+      return;
+    }
+    const file = files[0];
+    const start = 0;
+    const stop = file.size - 1;
+
+    const reader = new FileReader();
+    const that = this;
+    // If we use onloadend, we need to check the readyState.
+    reader.onloadend = function(evt) {
+      if (evt.target.readyState == FileReader.DONE) {
+        // DONE == 2
+        let layouts = { ...that.state.layouts };
+        layouts.lg = JSON.parse(evt.target.result).lg;
+        that.setState({ layouts });
+      }
+    };
+
+    const blob = file.slice(start, stop + 1);
+    reader.readAsBinaryString(blob);
+  };
+
+  saveJson = layouts => {
+    let dataStr = JSON.stringify(layouts);
+    let dataUri =
+      "data:application/json;charset=utf-8," + encodeURIComponent(dataStr);
+    let exportFileDefaultName = new Date().toISOString();
+
+    let linkElement = document.createElement("a");
```

Comment:
I think you could have avoid creating an extra element `a`. Even `saveJson` is useless if you add the attribute `href` directly on your `<a>` component inside render function. To have this to work, do something like:

```
<a
  href={ `data:application/json;charset=utf-8,${encodeURIComponent(dataStr)}`}
  layout={`layout ${exportFileDefaultName}.json`}
  className="btn btn-primary mb-3"
>
```

Reply:
Totally agree after tested and make sure it works! I removed saveJson and add href attribute direct to the element inside render function.